app/src/main/kotlin/com/exallium/todoapp/notedetail/NoteDetailPresenter.kt

```diff
@@ -0,0 +1,38 @@
+package com.exallium.todoapp.notedetail
+
+import com.exallium.todoapp.entities.Note
+import com.exallium.todoapp.mvp.BasePresenter
+import rx.Single
+import rx.SingleSubscriber
+import timber.log.Timber
+
+/**
+ * Presenter for Note Detail Screen
+ */
+class NoteDetailPresenter(val view: NoteDetailView, val model: NoteDetailModel) : BasePresenter<NoteDetailView>(view) {
+    lateinit var note: Note
+
+    override fun onViewCreated() {
+        if (view.isCached()) {
+            return
+        }
+        lookupNoteDetail(view.getNoteDetailId())
```

Comment:
I think what we want to do here is pass the view's bundle directly to the model.  The model will be injected with a ScreenBundleHelper instance, which can then read the note id, and either expose it, or, better yet, hide it and remove the param from getNoteById, and rename the function getNote, which will internally use the value in the bundle and properly handle nonexistence